server.go

```diff
@@ -340,6 +357,46 @@ func (srv *Server) ListenAndServe() error {
 		srv.started = true
 		unlock()
 		return srv.serveUDP(u)
+	case "doq", "doq4", "doq6":
+		if srv.TLSConfig == nil || (len(srv.TLSConfig.Certificates) == 0 && srv.TLSConfig.GetCertificate == nil) {
```

Comment:
Does does imply `WithInsecure()` doesn't work?

Reply:
I donno :D (disclaimer: I just copied this `if` from the TLS section)

Reply:
ok, would be nice to have that at some point